modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import {composeHooks, createElemPropsHook} from '@workday/canvas-kit-react/common';
+import {useComboboxInput} from '@workday/canvas-kit-react/combobox';
+import {useSelectModel} from './useSelectModel';
+
+/**
+ * `useSelectInput` extends {@link useComboboxInput useComboboxInput}  and adds type ahead functionality and select specific [keyboard support](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).
+ */
+export const useSelectInput = composeHooks(
+  useComboboxInput,
+  createElemPropsHook(useSelectModel)((model, ref) => {
+    const keySofar = React.useRef('');
+    const timer = React.useRef<ReturnType<typeof setTimeout>>();
+
+    React.useLayoutEffect(() => {
+      model.events.setWidth(model.state.targetRef.current?.clientWidth || 0);
+      // If there is no selected item and items exists we want to set the cursor to the first item in the array
+      if (model.state.selectedIds.length === 0 && model.state.items.length > 0) {
+        model.events.goTo({id: model.state.items[0].id});
+      } else {
+        // If the user wants an item selected by default by passing `initialSelectedId` we select that item
+        if (model.state.selectedIds.length > 0) {
+          const selectedItem = model.state.items.findIndex(
+            (item: {id: string}) => item.id === model.state.selectedIds[0]
+          );
+          model.events.goTo({id: model.state.items[selectedItem].id});
+          model.events.select({id: model.state.items[selectedItem].id});
+        }
+      }
+
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
This whole effect can prob go itno combobox. It's just selecting the item if there is one, if not,  moving the cursor to the first item and setting the widht of it to the traget.